test/integration/ssl_integration_test.cc

```diff
@@ -40,6 +44,7 @@ void SslIntegrationTest::TearDownTestCase() {
 }
 
 ServerContextPtr SslIntegrationTest::createUpstreamSslContext() {
+  static Stats::TestIsolatedStoreImpl upstream_stats_store;
```

Comment:
Why was this change made? I think this is fragile since I think we have tests that look at stats which might then not reset.